resolwe/storage/views.py

```diff
@@ -19,12 +19,31 @@
 from django.utils import timezone
 from django.views import View
 
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
 from resolwe.flow.models import Data
 from resolwe.flow.models.collection import Collection
+from resolwe.storage.connectors import connectors
 from resolwe.storage.connectors.baseconnector import BaseStorageConnector
 from resolwe.storage.models import FileStorage, ReferencedPath
 
 
+class TemporaryUploadCredentials(ViewSet):
+    """
+    Get the temporary credentials.
+    """
+
+    def list(self, request):
+        """Return the JSON representing temporary credentials."""
+        upload_connector = connectors.for_storage("upload")[0]
+        return Response(
+            upload_connector.temporary_credentials(str(request.user.id)),
```

Comment:
I think it is not the best idea to use the user's ID for the unique prefix. What would happen if someone uploads two files with the same name?

Reply:
It depends on the implementation in the web browser.

I believe it is better to namespace permissions (so nothing from other users can be effected) and that web browser just prefixes a random filename with user id.